src/components/select-field/BaseSelectField.js

```diff
@@ -188,16 +217,16 @@ class BaseSelectField extends React.Component<Props, State> {
         }
     };
 
-    handleBlur = () => {
+    handleBlur = (event: FocusEvent) => {
         const { isOpen } = this.state;
-        if (isOpen) {
+        if (isOpen && event.relatedTarget && event.relatedTarget.className !== 'search-input') {
```

Comment:
There is a call later on where this.searchInputRef.focus() is being called which means that handleBlur will be called here. We don't actually want to close the dropdown though because the user hasn't clicked outside of modal.